ambry-cloud/src/main/java/com/github/ambry/cloud/VcrReplicationManager.java

```diff
@@ -383,10 +413,27 @@ synchronized private void updateVcrHelix() {
       logger.warn("VCR Helix cluster update failed: ", e);
     } finally {
       isVcrHelixUpdateInProgress = false;
+      vcrUpdateDistributedLock.unlock();
     }
     logger.info("VCR Helix cluster update done.");
   }
 
+  private void checkAmbryHelixAndVcrHelixOnSync() {
+    boolean isSrcAndDstSync = false;
+    try {
+      String localDcZkStr = ((HelixClusterManager) clusterMap).getLocalDcZkConnectString();
+      isSrcAndDstSync = HelixVcrUtil.isSrcDestSync(localDcZkStr, clusterMapConfig.clusterMapClusterName,
+          cloudConfig.vcrClusterZkConnectString, cloudConfig.vcrClusterName);
+    } catch (Exception e) {
+      logger.warn("Ambry Helix and Vcr Helix sync check runs into exception: ", e);
+    }
+    if (!isSrcAndDstSync) {
+      logger.warn("Ambry Helix cluster and VCR helix cluster are not on sync");
+      // Raise alert on this metric
```

Comment:
should we do this check here before logging the metric

if(vcrHelixUpdateFuture == null && !isSrcAndDstSync) {
    // this means no update is scheduled or is in progress, and we still find that src and dest are out of sync
    vcrMetrics.vcrHelixNotOnSync.inc();
}

Reply:
good point. added.